addon/components/au-file-upload.js

```diff
@@ -0,0 +1,147 @@
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import { task } from 'ember-concurrency';
+import { guidFor } from '@ember/object/internals';
+
+export default class AuFileUploadComponent extends Component {
+  @service() store;
+  @service() fileQueue;
+  @tracked uploadErrorData = [];
+
+  constructor() {
+    super(...arguments);
+    this.queueName = `${guidFor(this)}-fileUploads`;
+  }
+
+  ////////
+  // Properties which may be overridden
+  ////////
+  get uploadingMsg() {
+    return `Bezig met het opladen van ${this.fileQueue.files.length} bestand(en). (${this.fileQueue.progress}%)`;
+  }
+
+  get title() {
+    return this.args.title || 'Bestanden toevoegen';
+  }
+
+  get helpTextDragDrop() {
+    return (
+      this.args.helpTextDragDrop ||
+      'Sleep de bestanden naar hier om toe te voegen'
+    );
+  }
+
+  get helpTextFileNotSupported() {
+    return (
+      this.args.helpTextFileNotSupported ||
+      'Dit bestandsformaat wordt niet ondersteund.'
+    );
+  }
+
+  get endPoint() {
+    return this.args.endPoint || '/files';
+  }
+
+  get modelName() {
+    return this.args.modelName || 'file';
+  }
+  get maxFileSizeMB() {
+    return this.args.maxFileSizeMB || 20;
+  }
+
+  ////////
+  // Error handling and introspection
+  ////////
+  get hasErrors() {
+    return this.uploadErrorData.length > 0;
+  }
+
+  // uploadFileTask = {};
+
+  @task({ enqueue: true, maxConcurrency: 3 })
+  *uploadFileTask(file) {
+    this.notifyQueueUpdate();
+    try {
+      const response = yield file.upload(this.endPoint, {
+        'Content-Type': 'multipart/form-data',
+      });
+      const uploadedFile = yield this.store.findRecord(
+        this.modelName,
+        response.body.data.id
+      );
+      return uploadedFile;
```

Comment:
The file uploader doesn't use Ember Data at all, but now it's a requirement to use this component (so technically it should also be listed as a dependency in the package.json file).

I think it makes more sense to return the file id to the action instead and let the app fetch the record if needed?

It's a bit more code when you use the component, but it's more flexible, requires less dependencies and things that need to be configurable.

Reply:
I'm beginning to see what it was a separate addon now, depending not only on ember-file-upload, but also on ember-concurrency and ember-data does make it quite large :). I like this suggestion though!